.github/workflows/test-311.yml

```
@@ -0,0 +1,49 @@
+name: Test
+
+on:
+  push:
+    paths-ignore:
+      - "docs/**"
+      - "*.md"
+
+  pull_request:
+    paths-ignore:
+      - "docs/**"
+      - "*.md"
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11-dev"]
```

`3.11.0-rc` is available in `actions/setup-python@v4`. Unless I am missing something, you don't have to use `deadsnakes/action`.

```suggestion
        python-version: ["3.11.0-rc - 3.11"]
```

This should provide any latest officially released version of Python 3.11.